src/core/Akka/Util/ByteString.cs

```diff
@@ -332,21 +332,21 @@ public ByteString Compact()
         /// </summary>
         /// <param name="index">index inside current <see cref="ByteString"/>, from which slicing should start</param>
         /// <param name="count">Number of bytes to fit into new <see cref="ByteString"/>.</param>
-        /// <returns></returns>
+        /// <exception cref="ArgumentOutOfRangeException">If index or count result in an invalid <see cref="ByteString"/>.</exception>
         public ByteString Slice(int index, int count)
         {
-            //TODO: this is really stupid, but previous impl didn't throw if arguments 
```

Comment:
@Horusiath 's comment here was the first sign that there was something weird going on inside this method, when I started investigating how our Petabridge.Cmd code could run out of memory. Sure enough - this method will produce arbitrary byte arrays regardless of what the `index` and `count` values are, which caused our looping mechanism for decoding messages to go haywire.

If someone is depending on this old behavior - sorry, but that's your problem now - we're going back to byte arrays meaning what they mean  starting in Akka.NET V1.5.